tests/test_main.py

```diff
@@ -129,6 +129,43 @@ def test_bad_limit(tmpdir):
     assert "Error: Invalid value" in stderr
 
 
+def test_rate_limit(tmpdir):
```

Comment:
Thanks for adding a test!

Reply:
I was originally trained as a developer, I do understand the value of tests.